sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FlatMapGroupsWithStateExec.scala

```diff
@@ -61,6 +61,10 @@ case class FlatMapGroupsWithStateExec(
 
   private val isTimeoutEnabled = timeoutConf != NoTimeout
   val stateManager = new FlatMapGroupsWithState_StateManager(stateEncoder, isTimeoutEnabled)
+  val watermarkPresent = child.output.exists {
```

Comment:
this is cleaner, but doesn't `eventTimeWatermark.isDefined` imply that the watermark is present?

Reply:
yes. agreed.

Reply:
Correction. No it is not. When watermark is not defined in the query, the eventTimeWatermark value is  Some(0)`